### ISSUE TITLE
Allow Taskgraph to store the result of a call

### ISSUE THREAD

**Issue by richpsharp** [natcap/taskgraph#15]
Add functionality to allow taskgraph to return the result of an asynchronous call in the current and future pre-calculated scenarios. Note some results may not be able to be pickled, so `add_task` should have an optional flag to disable this behavior.

**Comment by phargogh** [comment]
I am very excited about this change.

**Comment by richpsharp** [comment]
I've already done it! Just waiting on the other PR...
